storage/src/main/scala/coop/rchain/storage/regex/Multiplier.scala

```diff
@@ -16,27 +16,35 @@ private[regex] object Multiplier {
 
   def apply(multiplier: Int): Multiplier = new Multiplier(Some(multiplier), Some(multiplier))
 
-  def apply(symbol: String): Multiplier = symbol match {
-    case "*" => presetStar
-    case "?" => presetQuestion
-    case ""  => presetOne
-    case "+" => presetPlus
+  def parse(cs: CharSequence): Option[Multiplier] = tryParse(cs).map {
+    case (parsed, _) => parsed
   }
 
-  def parse(cs: CharSequence): Option[Multiplier] = {
-    val rxMinMax = "^\\{(\\d+)\\s*,\\s*(\\d+)}".r("min", "max")
-    val rxMin = "^\\{(\\d+)\\s*,\\s*}".r("min")
-
-    cs match {
-      case rxMinMax(min, max) => Some(Multiplier(min.toInt, max.toInt))
-      case rxMin(min)         => Some(Multiplier(Some(min.toInt), Multiplier.Inf))
-      case "*"                => Some(Multiplier.presetStar)
-      case "?"                => Some(Multiplier.presetQuestion)
-      case "+"                => Some(Multiplier.presetPlus)
-      case ""                 => Some(Multiplier.presetOne)
-      case _                  => None
+  private[this] val rxMinMax = "(^\\{\\s*(\\d+)\\s*,\\s*(\\d+)\\s*\\}).*".r("all", "min", "max")
+  private[this] val rxMin = "(^\\{\\s*(\\d+)\\s*,\\s*\\}).*".r("all", "min")
```

Comment:
Are you also going to support exactly n characters?  I.e. {min, max} {min,} {exactly}

Reply:
Scala question: can you use raw strings to reduce the backslash noise? Scala tells me:
```scala
scala> """(^\{\s*(\d+)\s*,\s*(\d+)\s*\}).*""" == "(^\\{\\s*(\\d+)\\s*,\\s*(\\d+)\\s*\\}).*"
res4: Boolean = true

scala> raw"(^\{\s*(\d+)\s*,\s*(\d+)\s*\}).*" == "(^\\{\\s*(\\d+)\\s*,\\s*(\\d+)\\s*\\}).*"
res5: Boolean = true
```
or does that break something?

Reply:
Thanks for a tip with """, replaced. Support for exactly {N} repetitions added. Decided to implement all cases with just one regex.